modules/test/playwright/tests/dynamic-data-mapping-form-web/main/formRules.spec.ts

```diff
@@ -54,96 +54,162 @@ test.beforeEach(({page}) => {
 	formBuilderFieldSettingsSidePanelPage,
 	formBuilderPage,
 	formBuilderSidePanelPage,
 	page,
 	rulesBuilderPage,
 }) => {
-	await formBuilderPage.goToNew();
+	await test.step('creating form and associated rule', async () => {
+		await formBuilderPage.goToNew();
 
-	await expect(formBuilderPage.newFormHeading).toBeVisible();
+		await expect(formBuilderPage.newFormHeading).toBeVisible();
 
-	await formBuilderPage.fillFormTitle('Form' + getRandomInt());
+		await formBuilderPage.fillFormTitle('Form' + getRandomInt());
 
-	await formBuilderSidePanelPage.addFieldByDoubleClick('Single Selection');
+		await formBuilderSidePanelPage.addFieldByDoubleClick(
+			'Single Selection'
+		);
 
-	await formBuilderFieldSettingsSidePanelPage.addOptions(2, 'single option ');
+		await formBuilderFieldSettingsSidePanelPage.addOptions(
+			2,
+			'single option '
+		);
 
-	await formBuilderSidePanelPage.clickBackButton();
+		await formBuilderSidePanelPage.clickBackButton();
 
-	await formBuilderSidePanelPage.addFieldByDoubleClick('Multiple Selection');
+		await formBuilderSidePanelPage.addFieldByDoubleClick(
+			'Multiple Selection'
+		);
 
-	await formBuilderFieldSettingsSidePanelPage.addOptions(
-		4,
-		'multiple option '
-	);
+		await formBuilderFieldSettingsSidePanelPage.addOptions(
+			4,
+			'multiple option '
+		);
 
-	await formBuilderSidePanelPage.advancedTab.click();
+		await rulesBuilderPage.rulesTab.click();
 
-	await formBuilderFieldSettingsSidePanelPage.fillMultiplePredefinedValues([
-		'multiple option 0',
-		'multiple option 3',
-	]);
+		await rulesBuilderPage.addElementsButton.click();
 
-	await page.waitForTimeout(1000);
+		await rulesBuilderPage.selectConditionLeftFormField('Single Selection');
 
-	await rulesBuilderPage.rulesTab.click();
+		await rulesBuilderPage.selectConditionOperator('Is Equal To');
 
-	await rulesBuilderPage.addElementsButton.click();
+		await rulesBuilderPage.selectConditionOperatorValueSource('Value');
 
-	await rulesBuilderPage.selectConditionLeftFormField('Single Selection');
+		await rulesBuilderPage.selectConditionRightFormField('single option 0');
 
-	await rulesBuilderPage.selectConditionOperator('Is Equal To');
+		await rulesBuilderPage.selectAction('Show');
 
-	await rulesBuilderPage.selectConditionOperatorValueSource('Value');
+		await page.getByRole('combobox').nth(5).click();
 
-	await rulesBuilderPage.selectConditionRightFormField('single option 0');
+		await page.getByRole('option', {name: 'Multiple Selection'}).click();
 
-	await rulesBuilderPage.selectAction('Show');
+		await page.getByRole('presentation').nth(4).click();
 
-	await page.getByRole('combobox').nth(5).click();
+		await page.getByRole('combobox').nth(6).click();
 
-	await page.getByRole('option', {name: 'Multiple Selection'}).click();
+		await page.getByRole('option', {name: 'Require'}).click();
 
-	await rulesBuilderPage.saveButton.click();
+		await page.getByRole('combobox').nth(7).click();
```

Comment:
Could you take a look at these [recommended locators](https://playwright.dev/docs/locators#quick-guide) to reduce the occurrences of the `.nth()` function?